File: sample-videochat-webrtc/sample-videochat-webrtc/Components/CallViewController/ZoomedView.m
//
//  ZoomedView.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 3/12/19.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

#import "ZoomedView.h"

@implementation ZoomedView

- (instancetype)initWithFrame:(CGRect)frame {
    
    self = [super initWithFrame:frame];
    if (self != nil) {
        self.backgroundColor = [UIColor colorWithWhite:0 alpha:1.0f];
        [self addGestureRecognizer:[[UITapGestureRecognizer alloc]
                                    initWithTarget:self
                                    action:@selector(didReceiveTap:)]];
    }
    return self;
}

- (void)setVideoView:(UIView *)videoView {
    
    if (_videoView != videoView) {
        _videoView = videoView;
        [videoView removeFromSuperview];
        videoView.frame = self.bounds;
        videoView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleBottomMargin;
        [self addSubview:videoView];
    }
}

- (void)didReceiveTap:(UITapGestureRecognizer *)__unused gestureRecognizer {
    if (self.didTapView != nil) {
        self.didTapView(self);
    }
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/sample-conference-videochat-swift/Components/Calls/CallViewController/ZoomedView.swift
//
//  ZoomedView.swift
//  sample-conference-videochat-swift
//
//  Created by Vladimir Nybozhinsky on 04.10.2018.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

class ZoomedView: UIView {
    //MARK: - Properties
    var videoView: UIView? {
        willSet{
            videoView?.removeFromSuperview()
        }
        
        didSet {
            guard let view = videoView else { return }
            
            addSubview(view)
            
            view.translatesAutoresizingMaskIntoConstraints = false
            view.leftAnchor.constraint(equalTo: leftAnchor).isActive = true
            view.rightAnchor.constraint(equalTo: rightAnchor).isActive = true
            view.topAnchor.constraint(equalTo: topAnchor).isActive = true
            view.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
        }
    }
    
    var didTapView: ((_ zoomedView: ZoomedView?) -> Void)?
    
    //MARK: - Life Cycle
    override init(frame: CGRect) {
        super.init(frame: frame)
        backgroundColor = UIColor(white: 0.0, alpha: 1.0)
        addGestureRecognizer(UITapGestureRecognizer(target: self,
                                                    action: #selector(didReceiveTap(_:))))
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    //MARK: - Actions
    
    @objc func didReceiveTap(_ sender: UITapGestureRecognizer?) {
        guard let didTapView = didTapView else { return }
        didTapView(self)
    }
}
